packer/plugin-getter/plugins.go

```diff
@@ -863,126 +860,103 @@ func (pr *Requirement) InstallLatest(opts InstallOptions) (*Installation, error)
 							return nil, errs
 						}
 
-						tempPluginPath := filepath.Join(os.TempDir(), fmt.Sprintf(
-							"packer-plugin-temp-%s%s",
-							random.Numbers(8),
-							opts.BinaryInstallationOptions.Ext))
-
-						// Save binary to temp so we can ensure it is really the version advertised
-						tempOutput, err := os.OpenFile(tempPluginPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+						tmpBinFileName := filepath.Join(os.TempDir(), expectedBinaryFilename)
 						if err != nil {
-							log.Printf("[ERROR] failed to create temp plugin executable: %s", err)
-							return nil, multierror.Append(errs, err)
+							err = fmt.Errorf("could not create temporary file to download plugin: %w", err)
+							errs = multierror.Append(errs, err)
+							return nil, errs
 						}
-						defer os.Remove(tempPluginPath)
+						defer func() {
+							tmpBinPath := tmpBinFileName
+							os.Remove(tmpBinPath)
+						}()
 
-						_, err = io.Copy(tempOutput, copyFrom)
+						tmpOutputFile, err := os.OpenFile(tmpBinFileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
 						if err != nil {
-							log.Printf("[ERROR] failed to copy uncompressed binary to %q: %s", tempPluginPath, err)
-							return nil, multierror.Append(errs, err)
+							err := fmt.Errorf("failed to create %s: %w", tmpBinFileName, err)
+							errs = multierror.Append(errs, err)
+							return nil, errs
+						}
+						if _, err := io.Copy(tmpOutputFile, copyFrom); err != nil {
+							err := fmt.Errorf("extract file: %w", err)
+							errs = multierror.Append(errs, err)
+							return nil, errs
 						}
 
-						// Not a problem on most platforms, but unsure Windows will let us execute an already
-						// open file, so we close it temporarily to avoid problems
-						_ = tempOutput.Close()
+						if _, err := tmpOutputFile.Seek(0, 0); err != nil {
+							err := fmt.Errorf("Error seeking beginning of binary file for checksumming: %w", err)
+							errs = multierror.Append(errs, err)
+							log.Printf("[WARNING] %v, ignoring", err)
+						}
 
-						desc, err := GetPluginDescription(tempPluginPath)
+						outputFileCS, err := checksum.Checksummer.Sum(tmpOutputFile)
 						if err != nil {
-							err := fmt.Errorf("failed to describe plugin binary %q: %s", tempPluginPath, err)
+							err := fmt.Errorf("failed to checksum binary file: %s", err)
 							errs = multierror.Append(errs, err)
-							continue
+							log.Printf("[WARNING] %v, ignoring", err)
```

Comment:
This is confusing.. If we failed to write the checksum we don't install the binary - even if it passed all the checks. Is that what we expect? 

I was confused by this.

Reply:
I added it back but will should remove the outputFile if we continue

Reply:
Ah yes definitely if we cannot write the checksum we should remove the plugin binary, it won't be loaded if the checksum is missing, maybe this should even be a break instead of a continue?
Like at this point the whole installation is valid on the remote, it's a local problem that prevented this failure so that's unlikely a blip, and I'm not certain continuing to the next candidate is a good move here.